include/zephyr/drivers/sensor.h

```diff
@@ -567,12 +611,43 @@ struct sensor_read_config {
 	static enum sensor_channel __channel_array_##name[] = {__VA_ARGS__};                       \
 	static struct sensor_read_config __sensor_read_config_##name = {                           \
 		.sensor = DEVICE_DT_GET(dt_node),                                                  \
+		.is_streaming = false,                                                             \
 		.channels = __channel_array_##name,                                                \
 		.count = ARRAY_SIZE(__channel_array_##name),                                       \
 		.max = ARRAY_SIZE(__channel_array_##name),                                         \
 	};                                                                                         \
 	RTIO_IODEV_DEFINE(name, &__sensor_iodev_api, &__sensor_read_config_##name)
 
+/**
+ * @brief Define a stream instance of a sensor
+ *
+ * Use this macro to generate a @ref rtio_iodev for starting a stream that's triggered by specific
+ * interrupts. Example:
+ *
+ * @code(.c)
+ * SENSOR_DT_STREAM_IODEV(imu_stream, DT_ALIAS(imu),
+ *     {SENSOR_TRIG_FIFO_WATERMARK, SENSOR_STREAM_DATA_INCLUDE},
+ *     {SENSOR_TRIG_FIFO_FULL, SENSOR_STREAM_DATA_NOP});
+ *
+ * int main(void) {
+ *   struct rtio_sqe *handle;
+ *   sensor_stream(&imu_stream, &rtio, NULL, &handle);
+ *   k_msleep(1000);
+ *   rtio_sqe_cancel(handle);
+ * }
+ * @endcode
+ */
+#define SENSOR_DT_STREAM_IODEV(name, dt_node, ...)                                                 \
+	static struct sensor_stream_trigger __trigger_array_##name[] = {__VA_ARGS__};              \
```

Comment:
Hi Yuval,

I met some problem when I integrated this PR. 
It would be better using _CONCAT instead of `##` when this macro is used by another macro.

Reply:
Done

Reply:
Thanks. Please also use `_CONCAT` in `SENSOR_DT_READ_IODEV`.